### PR TITLE
Add SAI Notification support for host_tx_ready

### PR DESCRIPTION
**Pull request by noaOrMlnx**

### CODE REVIEW

File: meta/Meta.cpp [hunk 1]
```diff
@@ -6434,6 +6434,55 @@ void Meta::meta_sai_on_nat_event(
...
+                    _In_ sai_object_id_t switch_id,
+                    _In_ sai_port_host_tx_ready_status_t host_tx_ready_status)
+{
+    SWSS_LOG_ENTER();
+
+    if (!sai_metadata_get_enum_value_name(
+            &sai_metadata_enum_sai_port_host_tx_ready_status_t,
+            host_tx_ready_status))
+    {
+        SWSS_LOG_WARN("port host_tx_ready value (%d) not found in sai_port_host_tx_ready_status_t",
```
Comment (Junchao-Mellanox) [inline meta/Meta.cpp]: do you want to return here and drop the notification is the value is invalid?

File: meta/Meta.cpp [hunk 2]
```diff
@@ -6434,6 +6434,55 @@ void Meta::meta_sai_on_nat_event(
...
+                host_tx_ready_status);
+    }
+    
+    auto ot = objectTypeQuery(port_id);
+    auto valid = false;
+    
+    switch (ot)
+    {
+        case SAI_OBJECT_TYPE_PORT:
+        // case SAI_OBJECT_TYPE_BRIDGE_PORT:
```
Comment (Junchao-Mellanox) [inline meta/Meta.cpp]: remove unused comment

File: meta/Meta.cpp [hunk 3]
```diff
@@ -6434,6 +6434,55 @@ void Meta::meta_sai_on_nat_event(
...
+            host_tx_ready_status))
+    {
+        SWSS_LOG_WARN("port host_tx_ready value (%d) not found in sai_port_host_tx_ready_status_t",
+                host_tx_ready_status);
+    }
+    
+    auto ot = objectTypeQuery(port_id);
+    auto valid = false;
+    
+    switch (ot)
```
Comment (Junchao-Mellanox) [inline meta/Meta.cpp]: do you need the switch/case here? how about if? 
```
valid =  (ot == SAI_OBJECT_TYPE_PORT);
```

File: meta/Meta.cpp [hunk 4]
```diff
@@ -6434,6 +6434,55 @@ void Meta::meta_sai_on_nat_event(
...
+            break;
+
+        default:
+            SWSS_LOG_ERROR("port_id %s has unexpected type: %s, expected PORT", //, BRIDGE_PORT or LAG
+                    sai_serialize_object_id(port_id).c_str(),
+                    sai_serialize_object_type(ot).c_str());
+            break;
+    }
+
+    if (valid && !m_oids.objectReferenceExists(port_id))
```
Comment (Junchao-Mellanox) [inline meta/Meta.cpp]: could you please help me understand why `!m_oids.objectReferenceExists(port_id)` is needed here? My understanding is like this: 
```
if port ref does not exist:
     insert port ref

if host ready key does not exist:
    create host key

```

File: meta/NotificationPortHostTxReadyEvent.cpp [hunk 5]
```diff
@@ -0,0 +1,73 @@
+#include "NotificationPortHostTxReadyEvent.h"
+
+#include "swss/logger.h"
+
+#include "sai_serialize.h"
+
+using namespace sairedis;
+
+// NotificationSwitchStateChange::NotificationSwitchStateChange(
```
Comment (Junchao-Mellanox) [inline meta/NotificationPortHostTxReadyEvent.cpp]: please remove unused comment

File: meta/NotificationPortHostTxReadyEvent.cpp [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+    SWSS_LOG_ENTER();
+
+    return m_switchId;
+}
+
+sai_object_id_t NotificationPortHostTxReady::getAnyObjectId() const
+{
+    SWSS_LOG_ENTER();
+
+    if (m_portId != SAI_NULL_OBJECT_ID)
```
Comment (Junchao-Mellanox) [inline meta/NotificationPortHostTxReadyEvent.cpp]: no need the if branch, suggest just:
```
return m_portId;
```

File: meta/SaiSerialize.cpp [hunk 7]
```diff
@@ -4593,6 +4639,31 @@ void sai_deserialize_port_oper_status_ntf(
...
+    try
+    {
+        j = json::parse(s);
+    }
+    catch (const std::exception&)
+    {
+        SWSS_LOG_ERROR("Recieved an exception after trying to parse %s", s.c_str());
+        return;
+    }
+    // auto count = (uint32_t)j.size();
```
Comment (Junchao-Mellanox) [inline meta/SaiSerialize.cpp]: please remove unused comment

File: meta/sai_serialize.h [hunk 8]
```diff
@@ -498,6 +513,9 @@ void sai_deserialize_free_bfd_session_state_ntf(
         _In_ uint32_t count,
         _In_ sai_bfd_session_state_notification_t* bfdsessionstate);
 
+void sai_deserialize_free_port_host_tx_ready_ntf(
```
Comment (Junchao-Mellanox) [inline meta/sai_serialize.h]: do we really need it?

File: syncd/scripts/syncd_init_common.sh [hunk 9]
```diff
@@ -233,6 +233,11 @@ config_syncd_mlnx()
     if [[ -f /tmp/sai_extra.profile ]]; then
         cat /tmp/sai_extra.profile >> /tmp/sai.profile
     fi
+
+    if [[ -f /$HWSKU_DIR/independent_mode_support.profile ]]; then
```
Comment (Junchao-Mellanox) [inline syncd/scripts/syncd_init_common.sh]: why need this?

File: meta/Meta.cpp (hunk 2 above)
Comment (noaOrMlnx) [reply meta/Meta.cpp]: @Junchao-Mellanox 
I should expect to receive only SAI_OBJECT_TYPE_PORT right?

File: syncd/scripts/syncd_init_common.sh (hunk 9 above)
Comment (noaOrMlnx) [reply syncd/scripts/syncd_init_common.sh]: got in here by mistake form another commit. removing

File: meta/NotificationFactory.cpp [hunk 10]
```diff
@@ -6,6 +6,7 @@
 #include "NotificationSwitchShutdownRequest.h"
 #include "NotificationSwitchStateChange.h"
 #include "NotificationBfdSessionStateChange.h"
+#include "NotificationPortHostTxReadyEvent.cpp"
```
Comment (stephenxs) [inline meta/NotificationFactory.cpp]: typo? usually, we include headers instead of sources.

**Pull request closed without merging**